.github/workflows/release.yml

```diff
@@ -1,69 +1,32 @@
 on: release
 name: Build binaries
+
 jobs:
-  release-linux-386:
-    name: release linux/386
+  release:
+    name: ${{ matrix.binaries }}
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        binaries: [release-linux-386, release-linux-amd64, release-darwin-386, release-darwin-amd64, release-windows-386, release-windows-amd64]
     steps:
-    - uses: actions/checkout@master
-    - name: compile and release
-      uses: ngs/go-release.action@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GOARCH: "386"
-        GOOS: linux
-  release-linux-amd64:
-    name: release linux/amd64
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@master
-    - name: compile and release
-      uses: ngs/go-release.action@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GOARCH: amd64
-        GOOS: linux
-  release-darwin-386:
-    name: release darwin/386
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@master
-    - name: compile and release
-      uses: ngs/go-release.action@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GOARCH: "386"
-        GOOS: darwin
-  release-darwin-amd64:
-    name: release darwin/amd64
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@master
-    - name: compile and release
-      uses: ngs/go-release.action@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GOARCH: amd64
-        GOOS: darwin
-  release-windows-386:
-    name: release windows/386
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@master
-    - name: compile and release
-      uses: ngs/go-release.action@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GOARCH: "386"
-        GOOS: windows
-  release-windows-amd64:
-    name: release windows/amd64
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@master
-    - name: compile and release
-      uses: ngs/go-release.action@v1.0.1
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        GOARCH: amd64
-        GOOS: windows
+      - uses: jungwinter/split@v1
+        id: split
+        with:
+          msg: ${{ matrix.binaries }}
+          seperator: '-'
+      - uses: actions/checkout@master
+      - name: compile and release
+        uses: ngs/go-release.action@v1.0.1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GOARCH: ${{ steps.split.outputs._2 }}
+          GOOS: ${{ steps.split.outputs._1 }}
+      - name: Upload to Docker 
+        uses: docker/build-push-action@v1
+        with:
+          repository: hugomd/cloudflare-ddns
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          tags: latest
```

Comment:
This is a really weird requirement, but I wonder if we could avoid pushing to `latest` altogether?

We made some breaking changes in [2.0.0](https://github.com/hugomd/cloudflare-ddns/releases/tag/2.0.0), so I've pinned the `latest` Docker tag to 1.1.1, so we avoid breaking the image for those currently using it. Although, we do display a warning advising migrating to the latest release..

Reply:
I removed the latest tag. It will only create a tag based on your release tag now. You can see that by comparing the [releases](https://github.com/tiemonl/cloudflare-ddns/releases) on my branch to the releases in [Docker Hub](https://hub.docker.com/r/tiemonl/cloudflare-ddns/tags).